snuba/api.py

```diff
@@ -347,7 +348,7 @@ def parse_and_run_query(dataset, request: Request, timer):
     if project_ids:
         request.query.add_conditions([('project_id', 'IN', project_ids)])
 
-    turbo = request.extensions['performance'].get('turbo', False)
+    turbo = request.query_settings.get('turbo', False)
     if not turbo:
         final, exclude_group_ids = get_projects_query_flags(project_ids)
```

Comment:
Yea, my plan was to move [this entire bit](https://github.com/getsentry/snuba/blob/f4f190352840fbe5ad5a9af6f1d5afb657d4c7c1/snuba/api.py#L347-L361) to the project processor. Will have to pass in `query_settings` into `process_query` too. And add a `add_final` method to `Query`.

I'll do it in this PR to make sure it works without too many issues.

Reply:
I am not sure I like the idea of coupling the query processor with settings. 
I would agree we did not define very well the semantics of settings yet, but in my opinion right now they are:
- request specific
- clickhouse specific

The second part specifically makes me think they would act on the clickhouse query and not on the snuba query.
Knowing the boundaries are still foggy, I would advise not to add this coupling at this stage since it would be easier not to add it now than to remove it later.
What about keeping the settings processing in the clickhouse query or at least at the same stage when the clickhouse query is processed and make it work whether or not project_id is there ?

Reply:
Hmmm, moving the turbo logic to ClickhouseQuery is going to be hard too because turbo affects final, and we need [final in stats](https://github.com/getsentry/snuba/blob/master/snuba/api.py#L326).

Maybe we should figure out stats before moving the turbo bit around?

Reply:
I don't think stats should drive this decision. Since we need "final" in stats we can always go around that problem either by:
- let util.raw_query retriev stats from the clickhouse query
or 
- recalculate final independently (would not recommend)

Reply:
Spoke to Filippo on Friday, and we decided that we do need to move the project specific code to the processor and pass in the settings.

Have also made it such that clickhouse query allows api to access the final variable for stats.